src/shared/Extensions/WebHookConfigurationExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Extension methods for <see cref="IConfiguration"/>.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal static class WebHookConfigurationExtensions
+    {
+        /// <summary>
+        /// Returns <see langword="true"/> if the configuration value with given <paramref name="key"/> is set to
+        /// 'true'; otherwise <see langword="false"/>.
+        /// </summary>
+        /// <param name="configuration">The <see cref="IConfiguration"/>.</param>
+        /// <param name="key">The key of the configuration value to evaluate.</param>
+        /// <returns><see langword="true"/> if the value is set to 'true'; otherwise <see langword="false"/>.</returns>
+        public static bool IsTrue(this IConfiguration configuration, string key)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            var value = configuration[key];
+            if (string.IsNullOrEmpty(value))
+            {
+                return false;
+            }
+
+            return bool.TryParse(value.Trim(), out var isSet) ? isSet : false;
```

Comment:
Do we really need this method at all? Seeing things like this in code makes my brain hurt.

It looks to my like TryParse already does what you want. https://msdn.microsoft.com/en-us/library/system.boolean.tryparse(v=vs.110).aspx

I also don't think we should be trimming whitespace around configuration values arbitrarily. Garbage in, garbage out.

Reply:
Some BCL `Try...(...)` methods `throw` on `null` or empty inputs. I mistakenly thought `bool.TryParse(...)` was one of those. So, yeah, I'll remove this class.